### DISCUSSION TITLE
Display - Profil für Generic NodeMCU 38 pin with SSD1306 benötigt

### DISCUSSION THREAD

**Discussion by silvermaster5** [tbnobody/OpenDTU#850] [Q&A]
Hallo zusammen,

wie geschrieben, habe ich einen ESP32 mit 38Pins und benötige daher das Profil für Generic NodeMCU 38 pin with SSD1306.
in der [nodemcu_esp32.json](https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles/nodemcu_esp32.json) sind im Moment nur Profile für ESP32 mit 32 Pins enthalten.

Hat jemand noch das Profil für den  Generic NodeMCU 38 pin with SSD1306 oder wie kann ich das selber erstellen, bzw was müsste ich in das json File eintragen?

Danke für eure Unterstützung!

**Comment by Juergen2453** [comment]
Schau mal dort. Ich habe die dort mal reingestellt.

https://discord.com/channels/984173303147155506/1060347063281401876/1097903129858678814

**Reply by silvermaster5** [reply]
Danke dir, aber ich bekomme die folgende Fehlermeldung wenn ich auf deinen Link klicke: 
![image](https://user-images.githubusercontent.com/131667082/234587751-484e382b-ae1b-4205-ac61-7fea92ed191c.png)
Kannst das du bitte mal prüfen?

**Comment by Juergen2453** [comment]
Bei mir geht der Link.
Vielleicht geht es so.
Der erste Eintrag war nur für mich.


[
    {
        "name": "JÃ¼rgen#8440 erstellt am 17.04.2023",
        "nrf24": {
            "miso": -1,
            "mosi": -1,
            "clk": -1,
            "irq": -1,
            "en": -1,
            "cs": -1
        },
        "cmt": {
            "clk": -1,
            "cs": -1,
            "fcs": -1,
            "sdio": -1,
            "gpio2": -1,
            "gpio3": -1
        },
        "display": {
            "type": -1,
            "data": -1,
            "clk": -1
        },
        "led": {
	    "led0": -1,
	    "led1": -1
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        }
    },
    {
        "name": "Generic NodeMCU 38 pin",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        }
    },
    {
        "name": "Generic NodeMCU 38 pin with NRF + CMT2300A",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "cmt": {
            "clk": 12,
            "cs": 27,
            "fcs": 26,
            "sdio": 14,
            "gpio2": -1,
            "gpio3": -1
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        }
    },
    {
        "name": "Generic NodeMCU 38 pin with SSD1306 0.9",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        },
        "display": {
            "type": 2,
            "data": 21,
            "clk": 22
        }
    },
    {
        "name": "Generic NodeMCU 38 pin with SSD1106 1.3",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        },
        "display": {
            "type": 3,
            "data": 21,
            "clk": 22
        }
    },
    {
        "name": "D1 Mini ESP32",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 17,
            "en": 16,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        }
    },
    {
        "name": "Olimex ESP32-POE",
        "nrf24": {
            "miso": 15,
            "mosi": 2,
            "clk": 14,
            "irq": 13,
            "en": 16,
            "cs": 5
        },
        "eth": {
            "enabled": true,
            "phy_addr": 0,
            "power": 12,
            "mdc": 23,
            "mdio": 18,
            "type": 0,
            "clk_mode": 3
        }
    },
    {
        "name": "Olimex ESP32-EVB",
        "nrf24": {
            "miso": 15,
            "mosi": 2,
            "clk": 14,
            "irq": 13,
            "en": 16,
            "cs": 17
        },
        "eth": {
            "enabled": true,
            "phy_addr": 0,
            "power": 12,
            "mdc": 23,
            "mdio": 18,
            "type": 0,
            "clk_mode": 0
        }
    },
    {
        "name": "Generic NodeMCU 30 pin",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 17,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        }
    },
    {
        "name": "WT32-ETH01",
        "nrf24": {
            "miso": 4,
            "mosi": 2,
            "clk": 32,
            "irq": 33,
            "en": 14,
            "cs": 15
        },
        "eth": {
            "enabled": true,
            "phy_addr": 1,
            "power": 16,
            "mdc": 23,
            "mdio": 18,
            "type": 0,
            "clk_mode": 3
        }
    },
    {
        "name": "LILYGO TTGO T-Internet-POE",
        "nrf24": {
            "miso": 2,
            "mosi": 15,
            "clk": 14,
            "irq": 34,
            "en": 12,
            "cs": 4
        },
        "eth": {
            "enabled": true,
            "phy_addr": 0,
            "power": -1,
            "mdc": 23,
            "mdio": 18,
            "type": 0,
            "clk_mode": 3
        }
    },
    {
        "name": "OpenDTU Fusion v1",
        "nrf24": {
            "miso": 48,
            "mosi": 35,
            "clk": 36,
            "irq": 47,
            "en": 38,
            "cs": 37
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
	"led": {
	    "led0": 17,
	    "led1": 18
        }
    },
    {
        "name": "OpenDTU Fusion v2 with NRF + CMT2300A",
        "nrf24": {
            "miso": 48,
            "mosi": 35,
            "clk": 36,
            "irq": 47,
            "en": 38,
            "cs": 37
        },
        "cmt": {
            "clk": 6,
            "cs": 4,
            "fcs": 21,
            "sdio": 5,
            "gpio2": 3,
            "gpio3": 8
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
	"led": {
	    "led0": 17,
	    "led1": 18
        }
    },
    {
        "name": "OpenDTU Fusion v1 with SSD1306 Display",
        "nrf24": {
            "miso": 48,
            "mosi": 35,
            "clk": 36,
            "irq": 47,
            "en": 38,
            "cs": 37
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
        "display": {
            "type": 2,
            "data": 2,
            "clk": 1
        }
    },
    {
        "name": "OpenDTU Fusion v1 with SH1106 Display",
        "nrf24": {
            "miso": 48,
            "mosi": 35,
            "clk": 36,
            "irq": 47,
            "en": 38,
            "cs": 37
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
        "display": {
            "type": 3,
            "data": 2,
            "clk": 1
        }
    }
]

**Comment by Juergen2453** [comment]
Oder so

[pin_mapping.zip](https://github.com/tbnobody/OpenDTU/files/11333618/pin_mapping.zip)

**Comment by silvermaster5** [comment]
Ok, danke dir. Ich hab das json eingespielt und die Hardware für ESP38 + SSD1306 ausgewählt. Die Konfigeinstellungen für Display sind jetzt:
![image](https://user-images.githubusercontent.com/131667082/234613844-73fe0570-3c8d-4a22-9f43-c3b94184f4c4.png)

Die Pins sind so gesteckt:
SCL auf GPIO22
SCA auf GPIO21
GND auf GND
VDO auf 5V

Müsste doch eigentlich laufen, oder hab ich einen Denkfehler?

**Reply by silvermaster5** [reply]
und mir ist aufgefallen, dass das Mapping für ESP32 §2 Pins und 38Pins identisch ist. 
![image](https://user-images.githubusercontent.com/131667082/234630437-b5e7f990-9fd1-433b-948b-8d7690048c4c.png)
Ist das korrekt?

**Comment by Juergen2453** [comment]
Im Bild links ist der ESP32 gemeint und rechts auch der ESP32 nur ich habe die Anzahl der Pins 38 geschrieben.
Den Text kann man ja frei wählen.

**Comment by silvermaster5** [comment]
heisst das also, dass die PIN Belegung bei ESP32 mit 32Pins und 38Pins identisch ist? ich hab alles gemacht wie beschrieben, aber mein Display bleibt leider dunkel.

**Comment by Juergen2453** [comment]
Es sind beides die selben. Nur die Bezeichnung hinter "name..." ist eine andere.
der Generic NodeMCU 32 with SSD1306 = ein ESP32 mit 38 pin
der Generic NodeMCU 38 pin with iSSD1306 0,9 = ein ESP32 mit 38 pin

welche Firmware-Version hast du ???

**Comment by Juergen2453** [comment]
Ein schneller Aufbau
Ich hoffe es hilft.

![20230427_175726](https://user-images.githubusercontent.com/116970574/234920494-ae21c18b-85aa-4ff8-b5bd-3969f904d2c5.jpg)

**Comment by silvermaster5** [comment]
Ich habe die neuste Software. Daran kann es nicht liegen. Ich habe aber noch einen anderen ESP und teste es mal damit
